pysat/_constellation.py

```diff
@@ -108,10 +140,119 @@ def __str__(self):
                 output_str += "{:d} '{:s}' '{:s}' '{:s}' '{:s}'\n".format(
                     i, inst.platform, inst.name, inst.tag, inst.inst_id)
         else:
-            output_str += "No loaded Instruments\n"
+            output_str += "No assigned Instruments\n"
+
+        # Display loaded data
+        output_str += '\n\nLoaded Data Statistics\n'
+        output_str += '----------------------\n'
+        if not self.empty:
+            output_str += 'Date: ' + self.date.strftime('%d %B %Y') + '\n'
+            output_str += '{:s}time range: '.format(
+                "Common " if self.common_index else "Full ")
+            output_str += self.index[0].strftime('%d %B %Y %H:%M:%S')
+            output_str += ' --- '
+            output_str += self.index[-1].strftime('%d %B %Y %H:%M:%S\n')
+            output_str += 'Number of variables: {:d}\n'.format(
+                len(self.variables))
+
+            output_str += '\nVariable Names:\n'
+            output_str += utils._core.fmt_output_in_cols(self.variables)
+        else:
+            output_str += 'No loaded data.\n'
 
         return output_str
 
+    # -----------------------------------------------------------------------
+    # Define all hidden methods
+
+    def _empty(self, all_inst=True):
+        """Boolean flag reflecting lack of data
+
+        Parameters
+        ----------
+        all_inst : bool
+            Require all instruments to have data for the emtpy flag to be
+            False, if True.  If False, the empty flag will be False if any
+            instrument has data. (default=True)
+
+        Returns
+        -------
+        eflag : bool
+            True if there is no Instrument data, False if there is data
+            in at least one/all Instruments.
+
+        """
+        eflags = [inst.empty for inst in self.instruments]
+
+        if len(eflags) == 0:
+            eflag = True
+        elif all_inst:
+            eflag = np.any(eflags)
+        else:
+            eflag = np.all(eflags)
+
+        return eflag
+
+    def _index(self):
+        """Returns a common time index for the loaded data
+
+        Returns
+        -------
+        cindex : pds.Series
+            Series containing a common time index for the Instrument data
+
+        """
+        if len(self.instruments) == 0 or self.empty:
+            cindex = pds.Index([])
+        else:
+            stime = None
+            etime = None
+            out_res = None
+
+            for inst in self.instruments:
+                if stime is None:
+                    # Initialize the start and stop time
+                    stime = inst.index[0]
+                    etime = inst.index[-1]
+
+                    # If desired, determine the resolution
+                    if self.index_res is None:
+                        if inst.index.freq is None:
+                            out_res = (inst.index[1:] - inst.index[:-1]).mean()
```

Comment:
What if an instrument say
- switched sample rate during the loaded data
- had a variety of gaps which would increase the mean past the actual sample rate 

I noticed there is a `calc_freq` in utils.time based off the minimum time difference. I think you wrote that. Would that work here?

Reply:
Probably! 🦊  I will go take a look at that.

Reply:
Added two new functions to fix this and other problems I noted when going back and looking at this more closely!  Will re-request a review after I finish the unit tests.